src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -183,6 +209,141 @@ private void validateInconsistentPeriodFiling(Debtors debtors, Errors errors) {
             addInconsistentDataError(errors, PREVIOUS_TOTAL_PATH);
         }
     }
-}
 
+    /**
+     * @inheritDoc
+     */
+
+    @Override
+    public Errors crossValidate(Errors errors, HttpServletRequest request, String CompanyAccountsId,
+                                Debtors debtors) throws DataException {
+
+        crossValidateCurrentPeriod(errors, request, debtors, CompanyAccountsId);
+        crossValidatePreviousPeriod(errors, request, CompanyAccountsId, debtors);
+
+        return errors;
+    }
+
+    private void crossValidatePreviousPeriod(Errors errors, HttpServletRequest request,
+                                             String CompanyAccountsId,
+                                             Debtors debtors) throws DataException {
+
+        String previousPeriodId = previousPeriodService.generateID(CompanyAccountsId);
+
+        // get previous period object
+        ResponseObject<PreviousPeriod> previousPeriodResponseObject;
+
+        try {
+            previousPeriodResponseObject =
+                previousPeriodService.findById(previousPeriodId, request);
+        } catch (MongoException e) {
+
+            throw new DataException(e.getMessage(), e);
+        }
+
+        // if note is null and balance sheet value isn't add error
+        if (isDebtorsNotePreviousTotalNull(debtors) &&
+            (!isPreviousPeriodBalanceSheetDebtorsNull(previousPeriodResponseObject))) {
+
+            addError(errors, previousBalanceSheetNotEqual, PREVIOUS_TOTAL_PATH);
+        }
+
+        // if balance sheet value is null and note value isn't add error
+        if (previousPeriodResponseObject.getData() == null &&
+            !isDebtorsNotePreviousTotalNull(debtors)) {
+
+            addError(errors, previousBalanceSheetNotEqual, PREVIOUS_TOTAL_PATH);
+        } else {
+            if (isPreviousPeriodBalanceSheetDebtorsNull(previousPeriodResponseObject) &&
+                (!isDebtorsNotePreviousTotalNull(debtors))) {
+
+                addError(errors, previousBalanceSheetNotEqual, PREVIOUS_TOTAL_PATH);
+            }
+
+            // if neither are null and values do not match add error
+            if (!isPreviousPeriodBalanceSheetDebtorsNull(previousPeriodResponseObject) &&
+                (!isDebtorsNotePreviousTotalNull(debtors))
+
+                && (!debtors.getPreviousPeriod().getTotal().equals(
+                previousPeriodResponseObject.getData().getBalanceSheet().getCurrentAssets()
+                                            .getDebtors()))) {
+
+                addError(errors, previousBalanceSheetNotEqual, PREVIOUS_TOTAL_PATH);
+            }
+        }
+    }
+
+    private void crossValidateCurrentPeriod(Errors errors, HttpServletRequest request,
+                                            Debtors debtors,
+                                            String CompanyAccountsId) throws DataException {
+
+        String currentPeriodId = currentPeriodService.generateID(CompanyAccountsId);
```

Comment:
We could do the same as before with the `crossValidateCurrentPeriod` , to broke it down in smaller methods

Reply:
new method created to getCurrentPeriodResponseObject in b39ab32